Block/Sidebar.php

```diff
@@ -206,59 +204,52 @@ public function getChildCategoryView($category, $html = '', $level = 1)
     /**
      * Retrieve subcategories
      * DEPRECATED
-	 *
+     *
      * @param $category
      *
-     * @return array
+     * @return array|Collection|Category[]
      */
-	 
-    public function getSubcategories($category)
+
+    public function getSubcategories(Category $category)
     {
-        if ( $this->categoryFlatConfig->isFlatEnabled() && $category->getUseFlatResource() )
-        {
+        if ($this->categoryFlatConfig->isFlatEnabled() && $category->getUseFlatResource()) {
             return (array)$category->getChildrenNodes();
         }
 
         return $category->getChildrenCategories();
     }
-	
 
     /**
      * Get current category
      *
-     * @param \Magento\Catalog\Model\Category $category
+     * @param Category $category
      *
-     * @return Category
      */
-    public function isActive($category)
+    public function isActive(Category $category): bool
     {
         $activeCategory = $this->_coreRegistry->registry('current_category');
         $activeProduct  = $this->_coreRegistry->registry('current_product');
 
-        if ( !$activeCategory )
-        {
+        if (!$activeCategory) {
 
             // Check if we're on a product page
-            if ( $activeProduct !== null )
-            {
+            if ($activeProduct !== null) {
                 return in_array($category->getId(), $activeProduct->getCategoryIds());
             }
 
             return false;
         }
 
         // Check if this is the active category
-        if ( $this->categoryFlatConfig->isFlatEnabled() && $category->getUseFlatResource() AND
+        if ($this->categoryFlatConfig->isFlatEnabled() && $category->getUseFlatResource() and
```

Comment:
If we change this, we should probably change to `&&`?

Reply:
Indeed